theano/tensor/extra_ops.py

```diff
@@ -450,7 +457,48 @@ def bincount(x, weights=None, minlength=None):
 
     .. versionadded:: 0.6
     """
-    return BinCountOp(minlength=minlength)(x, weights)
+    compatible_type = ('int8', 'int16', 'int32', 'int64',
+                       'uint8', 'uint16', 'uint32', 'uint64')
+    
+    if x.dtype not in compatible_type:
+        raise TypeError("Inputs dtype must be an integer.")
+
+    # Some dtypes are not supported by numpy's implementation of bincount.
+    # Until another one is available, we should fail at graph construction
+    # time, not wait for execution.
+    int_bitwidth = theano.gof.python_int_bitwidth()
+    if int_bitwidth == 64:
+        numpy_unsupported_dtypes = ('uint64',)
+    if int_bitwidth == 32:
+        numpy_unsupported_dtypes = ('uint32', 'int64', 'uint64')
+        
+    intp_bitwidth = theano.gof.local_bitwidth()
+    if intp_bitwidth == 32:
+        out_type = basic.ivector()
+    elif intp_bitwidth == 64:
+        out_type = basic.lvector()
+
+    if x.dtype in numpy_unsupported_dtypes:
+        raise TypeError(
+            ("Input dtypes %s are not supported by numpy.bincount, "
+             % numpy_unsupported_dtypes), x.dtype)
+
+    if x.ndim != 1:
+        raise TypeError("Inputs must be of dimension 1.")
+
+    max_value = x.max() + 1
+
+    if minlength is not None:
+        max_value = theano.tensor.maximum(max_value, minlength)
+
+    if weights is None:
+        out = theano.tensor.zeros([max_value], dtype=out_type.dtype)
+        out = theano.tensor.inc_subtensor(out[x], 1)
+    else:
+        out_type = basic.dvector()
+        out = theano.tensor.zeros([max_value], dtype=out_type.dtype)
+        out = theano.tensor.inc_subtensor(out[x], weights)
+    return out
```

Comment:
In general, a few variables need to be renamed and a few messages/comments updated because this function is not wrapping the numpy implementation.